### PR TITLE
Feature/em 84 create code climate projects summary

### PR DESCRIPTION
**Pull request by mrubi-rootstrap**
## What does this PR do?

- It implements the [query to CodeClimate projects summary](https://rootstrap.atlassian.net/jira/software/projects/EM/boards/42?selectedIssue=EM-84)
- It adds the field CodeClimateProjectMetric.snapshot_time
- It creates the ProjectsSummaryService class to query the projects CodeClimate summary
- It creates the ProjectsSummary class to gather the information about projects

- It implements only the backend integration, the frontend integration is left to a different PR
- It still remains to implement the filter by `department`. That change has not been merged into `develop` yet, once it is this PR will add that missing feature

### CODE REVIEW

File: app/services/code_climate/projects_summary_service.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      return if code_climate_metrics.empty?
+
+      code_climate_metrics.map(&:open_issues_count).sum / code_climate_metrics.size
+    end
+
+    def ratings
+      return if code_climate_metrics.empty?
+
+      hash = Hash.new { |h, rate| h[rate] = 0 }
+      code_climate_metrics.each_with_object(hash) do |code_climate_metrics, ratings|
```
Comment (horacio) [inline app/services/code_climate/projects_summary_service.rb:48]:
```suggestion
      code_climate_metrics.each_with_object(Hash.new(0)) do |code_climate_metrics, ratings|
```

File: app/services/code_climate/projects_summary_service.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      return [] if department != 'web' # waiting for department to be merged into develop
+
+      @code_climate_metrics ||=
+        technologies.empty? ? metrics_in_any_technology : metrics_in_given_technologies
+    end
+
+    def metrics_in_given_technologies
+      CodeClimateProjectMetric
+        .joins(:project)
+        .where('snapshot_time >= ? AND projects.lang IN (?)', from, technologies)
```
[the remark this answers is not in the document]
Comment (mrubi-rootstrap) [reply app/services/code_climate/projects_summary_service.rb:63]: Done it. Thanks!

File: app/services/code_climate/projects_summary_service.rb (hunk 1 above)
Comment (mrubi-rootstrap) [reply app/services/code_climate/projects_summary_service.rb]: Done it. Thanks!

File: app/services/code_climate/projects_summary_service.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        metrics_in_department.where('snapshot_time >= ?', from)
+      else
+        metrics_in_department
+      end
+    end
+
+    def metrics_in_department
+      CodeClimateProjectMetric
+        .joins(:project)
+        .where('department_id = ?', department)
```
Comment (hvilloria) [inline app/services/code_climate/projects_summary_service.rb:73]:
```suggestion
        .where(department_id: department.id)
```

Comment (mrubi-rootstrap) [reply app/services/code_climate/projects_summary_service.rb]: Done it. Thanks!

**Pull request merged**